packages/table/components/Table.tsx

```diff
@@ -56,9 +56,17 @@ export class Table<T> extends React.PureComponent<TableProps, TableState> {
       children: Array<React.ReactElement<ColumnProps>>,
       width: number
     ): number[] => {
-      const totalColumns: number = children.length;
       let remainingWidth: number = width;
-      return children.map((child, currentIndex) => {
+      const clampWidth = (
```

Comment:
`getColumnSizes` has to be refactored, it grows out of bounds. There are several functions inside that can be extracted and the name is misleading, we then pass the result as `<MultiGrid columnWidth={getColumnSize} />`.
While name change is out of scope I would like to ask to consider refactoring the implementation.